### DISCUSSION TITLE
Need to have a field titled 'name'

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#318] [Q&A]
I was just struggling to figure out why I was getting a validation error for an "incomplete model" and realized that I was using a field called 'title' rather than 'name' in my model definition which was causing the issue.

Question:  Do we have this documented somewhere?  If not, we should.  I think I understand why this is needed, but it isn't obvious.

I just wanted to capture it here so we don't lose it.

**Comment by Coffee2Bits** [comment]
I think `title` would be an acceptable alternate to `name`. Whatever the name for the field, we do rely on unique names for the various definitions/models since we use those names as references between models -- at least for now. 

We can update the documentation to make sure that all definitions/models in the documentation adhere to the correct structure with a name field, but the field `name` is listed as a required field for any `data:` definition in the core spec. Perhaps we need to revisit the error and make it clearer.

Other than documentation changes and revisiting the error message, is there anything that might have made this more obvious? Once we're able to hook up the AaC LSP diagnostics with the validator results then the IDE would immediately bring your attention to the missing required field, `name` in your definition.

Here is a quick overview of the [LSP diagnostics](https://code.visualstudio.com/api/language-extensions/programmatic-language-features#provide-diagnostics). The goal for this would be to have validation running automatically on all AaC files via the VSCode extension, then any findings from the validation will be reported to the user via the IDE by leveraging LSP diagnostics.

**Reply by jondavid-black** [reply]
After thinking about it a bit more, it appears the name field is only required on a data item that can act as a root type.  At first (i.e. when I wrote this question) I thought every data item had to have a field called `name`.  Now that I've connected this to how we handle root types, this makes sense but may not be obvious to everyone at first.  I think we can just capture it in documentation for plug-in developers in the future.

**Comment by Coffee2Bits** [comment]
Revisiting this discussion; it sounds like this is more of an issue with making the spec and definitions' schemas visible to the user, something that we are addressing via the visual editor and the LSP features. 

With the new user documentation, it should be more apparent to users how the schema and definitions align, and which structures (e.g. `schema`, `model`, etc) require which fields. 

Given that this issue was written before the new user documentation, and that we're addressing the visibility of the schema via the new tooling, are there any additional items that we are overlooking that would ease this user experience friction?
